### PR TITLE
Add Fortinet Fortios Parser support

### PR DESCRIPTION
**Pull request by jeffkala**
Initial parser support for fortinet.

### CODE REVIEW

**Comment by itdependsnetworks** [comment]
can you add tests in `netutils/tests/unit/mock/config/compliance/` as well?

File: tests/unit/mock/config/parser/fortinet_fortios/fortios_full_received.py [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+    ConfigLine(config_line="    set admin-hsts-max-age 15552000", parents=("config system global",)),
+    ConfigLine(config_line="    set admin-https-pki-required disable", parents=("config system global",)),
+    ConfigLine(config_line="    set admin-https-redirect enable", parents=("config system global",)),
+    ConfigLine(
+        config_line="    set admin-https-ssl-versions tlsv1-1 tlsv1-2 tlsv1-3", parents=("config system global",)
+    ),
+    ConfigLine(config_line="    set admin-lockout-duration 60", parents=("config system global",)),
+    ConfigLine(config_line="    set admin-lockout-threshold 3", parents=("config system global",)),
+    ConfigLine(config_line="    set admin-login-max 100", parents=("config system global",)),
+    ConfigLine(config_line="end", parents=()),
```
Comment (itdependsnetworks) [inline tests/unit/mock/config/parser/fortinet_fortios/fortios_full_received.py:16]: Is this correct? Should `end` keep context of a parent of `config system global` here?

Comment (itdependsnetworks) [reply tests/unit/mock/config/parser/fortinet_fortios/fortios_full_received.py:16]: https://docs.fortinet.com/document/fortigate/6.0.0/cli-reference/252305/command-syntax

Comment (jeffkala) [reply tests/unit/mock/config/parser/fortinet_fortios/fortios_full_received.py:16]: Will read the guide closer.  I had thought it should be its own 'parent' but I suppose it is closing out a specific parent.  I work to get that fixed.  Also will add more tests.

**Comment by itdependsnetworks** [comment]
I think this looks good, but I am really coming to that conclusion based on the tests. @jmcgill298 can you take a look?

**Review by jmcgill298 — APPROVED** [review]
LGTM

**Comment by jeffkala** [comment]
Testing in a customer env this morning, having a few issues so working through those now and will update the PR/test.  Seems like for `end` and `next` lines I'm going to have to do some further cleanup or some errors happen.

**Comment by jeffkala** [comment]
@itdependsnetworks working now, tested locally as well.

File: netutils/config/parser.py [hunk 2]
```diff
@@ -695,12 +695,36 @@ def __init__(self, config):
...
+
     def _parse_out_offending(self, config):  # pylint: disable=no-self-use
         """Preprocess out strings that offend the normal spaced configuration syntax.
 
         Args:
             config (str): full config as a string.
         """
+        # This will grab everything between quotes after the 'set buffer' sub-command.
+        # Its explicitly looking for "\n to end the captured data.  This is to support html
+        # data that is supported in Fortinet config with double quotes within the html.
```
Comment (itdependsnetworks) [inline netutils/config/parser.py:727]: Still need to document in someway that get to read the docs that we will not an html syntax that somehow starts the line with a double quote.

File: docs/source/netutils/configs/index.rst [hunk 3]
```diff
@@ -9,4 +9,11 @@ Configs
...
-    :members:
\ No newline at end of file
+    :members:
+
+Edge Cases
+==============
+
+Fortinet Fortios Parser
+-----------------------
+- In order to support html blocks that exist in Fortios configurations, some preprocessing is executed, this is a regex that specifically grabs everything between quotes after the 'set buffer' sub-command. Its explicitly looking for double quote followed by a newline ("\n) to end the captured data.  This support for html data will not support any other html that doesn't follow this convention.
```
Comment (qduk) [inline docs/source/netutils/configs/index.rst:19]:
```suggestion
- In order to support html blocks that exist in Fortios configurations, some preprocessing is executed, this is a regex that specifically grabs everything between quotes after the 'set buffer' sub-command. It's explicitly looking for double quote followed by a newline ("\n") to end the captured data.  This support for html data will not support any other html that doesn't follow this convention.
```

Comment (jeffkala) [reply docs/source/netutils/configs/index.rst:19]: Will fix the `It's` but the second is actually intended.

**Review by qduk — APPROVED** [review]

**Pull request merged**